sysl2/sysl/seqs.go

```diff
@@ -19,12 +19,23 @@ type sequenceDiagParam struct {
 	title      string
 	blackboxes map[string]*Upto
 	appName    string
+	groupboxes map[string]StrSet
 }
 
 func generateSequenceDiag(m *sysl.Module, p *sequenceDiagParam) (string, error) {
 	w := MakeSequenceDiagramWriter(true, "skinparam maxMessageSize 250")
 	v := MakeSequenceDiagramVisitor(p.AppLabeler, p.EndpointLabeler, w, m, p.appName)
 	e := MakeEndpointCollectionElement(p.title, p.endpoints, p.blackboxes)
+
+	const color = "#LightBlue"
+	for boxname, appset := range p.groupboxes {
```

Comment:
This is wrong. This will output all apps that are in the module. 
We only want apps that are part of the sequence diagram

Build group boxes when you are building the unique var for each app. You only build var for an app that will show up in the seqs diagram.
At that time look up if the app has an attribute of interest (value specified by the `group by`) and put them in the map.

Reply:
Taken care